apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java

```diff
@@ -0,0 +1,50 @@
+package gov.cdc.nbs.questionbank.page;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import gov.cdc.nbs.questionbank.entity.WaTemplate;
+import gov.cdc.nbs.questionbank.entity.repository.WaTemplateRepository;
+import gov.cdc.nbs.questionbank.page.response.PageStateResponse;
+import gov.cdc.nbs.questionbank.page.util.PageConstants;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class PageStateChanger {
+
+	@Autowired
+	private WaTemplateRepository templateRepository;
+
+	public PageStateResponse savePageAsDraft(Long id) {
+		PageStateResponse response = new PageStateResponse();
+		try {
+			Optional<WaTemplate> result = templateRepository.findById(id);
+			if (result.isPresent()) {
+				WaTemplate template = result.get();
+				template.setTemplateType("Draft");
+				template.setPublishVersionNbr(null);
+				template.setPublishIndCd('F');
+				templateRepository.save(template);
+				response.setMessage(template.getId() + PageConstants.SAVE_DRAFT_SUCCESS);
+				response.setStatus(HttpStatus.OK);
+				response.setTemplateId(template.getId());
+
+			} else {
+				response.setMessage(PageConstants.SAVE_DRAFT_FAIL);
+				response.setStatus(HttpStatus.NOT_FOUND);
+				response.setTemplateId(id);
+			}
+		} catch (Exception e) {
+			response.setMessage(e.getMessage());
+			response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR);
+			response.setTemplateId(id);
+
+		}
+		return response;
```

Comment:
Legacy does not allow converting an existing "published" page back into a draft. It creates a new page from the existing one without modifying the published page. (Plz double check and validate this though).

Workflow that I have seen:
1. Create a new page (`template_type` is set to "Draft", `publish_version_nbr` of NULL)
2. Publish that page (`template_type` is set to "Published")
3. Create a draft from a published page - creates a new entry in the `wa_template` table with status of "Draft" and a `publish_version_nbr` of 0, the published page has its `template_type` set to `Published With Draft`.

The query that fetches the pages to display their summaries in table format only selects `template_type` of "Draft" or "Published", this prevents the duplicates from showing up in the summary table.

Reply:
@mpeels  sure I will confirm

Reply:
Hi @mpeels  updated please take a look:

<img width="973" alt="Screenshot 2023-08-02 at 2 29 18 PM" src="https://github.com/CDCgov/NEDSS-Modernization/assets/119972899/08e75743-8d91-4c3a-a7ca-dda6b8700a37">

<img width="1069" alt="Screenshot 2023-08-02 at 2 47 58 PM" src="https://github.com/CDCgov/NEDSS-Modernization/assets/119972899/8807d6a3-3eb1-4c8b-80a5-949c10ed1c57">